controllers/v1/programs.js

```diff
@@ -3,13 +3,354 @@
+   * @returns {JSON} - created program document.
+   */
+
+    async create(req) {
+      return new Promise(async (resolve, reject) => {
+        try {
+          let programCreationData = await programsHelper.create(
+            req.body,
+            req.userDetails.userId,
+            true
+          );
+          
+          return resolve({
+            message : CONSTANTS.apiResponses.PROGRAMS_CREATED,
+            result : _.pick(programCreationData,["_id"])
+          });
+  
+        }
+        catch (error) {
+          reject({
+            status: error.status || HTTP_STATUS_CODE.internal_server_error.status,
+            message: error.message || HTTP_STATUS_CODE.internal_server_error.message,
+            errorObject: error
+          })
+        }
+      })
+    }
+
+
+  /**
+   * Update program.
+   * @method
+   * @name update
+   * @param {Object} req - requested data.
+   * @apiParamExample {json} Request-Body:
+   * {
+      "externalId" : "PROGID01",
+      "name" : "DCPCR School Development Index 2018-19",
+      "description" : "DCPCR School Development Index 2018-19",
+      "isDeleted" : false,
+      "resourceType" : [ 
+          "program"
+      ],
+      "language" : [ 
+          "English"
+      ],
+      "keywords" : [],
+      "concepts" : [],
+      "userId":"a082787f-8f8f-42f2-a706-35457ca6f1fd",
+      "imageCompression" : {
+          "quality" : 10
+      },
+      "components" : [ 
+          "5b98fa069f664f7e1ae7498c"
+      ],
+      "scope" : {
+          "entityType" : "state",
+          "entities" : ["bc75cc99-9205-463e-a722-5326857838f8","8ac1efe9-0415-4313-89ef-884e1c8eee34","5f33c3d85f637784791cd830"],
+          "roles" : ["HM","BEO"]
+      },
+      "requestForPIIConsent" : true
+  }
+
+   * @param {Object} 
+   * @returns {JSON} - 
+   */
+
+  async update(req) {
+    return new Promise(async (resolve, reject) => {
+      try {
+
+        let programUpdationData = await programsHelper.update(
+          req.params._id,
+          req.body,
+          req.userDetails.userId,
+          true
+        );
+        
+        programUpdationData.result = programUpdationData.data;
+        return resolve(programUpdationData);
+
+      }
+      catch (error) {
+        reject({
+          status: error.status || HTTP_STATUS_CODE.internal_server_error.status,
+          message: error.message || HTTP_STATUS_CODE.internal_server_error.message,
+          errorObject: error
+        })
+      }
+    })
+  }
+
+  /**
+   * Details of the program
+   * @method
+   * @name details
+   * @param {Object} req - requested data.
+   * @param {String} req.params._id - program id.
+   * @returns {Array} Program scope roles.
+   */
+
+      async details(req) {
+      return new Promise(async (resolve, reject) => {
+        try {
+  
+          let programData = await programsHelper.details(
+            req.params._id
+          );
+  
+          programData["result"] = programData.data;
```

Comment:
@Prajwal17Tunerlabs can you move this to helper and send the result from helper itself